src/main/java/com/google/devtools/build/lib/actions/FileArtifactValue.java

```diff
@@ -172,6 +173,17 @@ protected boolean couldBeModifiedByMetadata(FileArtifactValue lastKnown) {
     return true;
   }
 
+  /**
+   * Optional symlink target path.
+   *
+   * <p>If present, this artifact is a copy of another artifact. It is materialized in the local
+   * filesystem as a symlink to the original artifact, whose contents live at this location. This is
+   * used to implement zero-cost copies of remotely stored artifacts.
+   */
+  public Optional<PathFragment> getSymlinkTargetExecPath() {
```

Comment:
This is out of sync with the `SourceFileArtifactValue`. If that is intended to be short lived, maybe let's move that to a separate interface and case in remote. If that is long lived, it may be worth to keep existing code in sync.

Reply:
Makes sense, please consider naming that more specifically to make sure that is clear why this is separate, e.g. `getSymlinkTargetPathForLocalPrefetching` (I am confident you can come up with something better than that).

Reply:
I changed the name to "materialization" which I think is distinctive enough to cause readers to pay attention; it also avoids a potential confusion with unresolved symlinks, which are entirely separate. (I couldn't bring myself to include all of "symlink", "exec path" and "materialization"/"prefetching" in the name, sorry; it's already uncomfortably long as it is.)